docs/guides/external-mcp-server.md

```diff
@@ -0,0 +1,299 @@
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: github-mcp-external
+  namespace: mcp-test
+spec:
+  parentRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: mcp-gateway
+    namespace: gateway-system
+  hostnames:
+  - api.githubcopilot.com  # must match the Gateway listener hostname
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /mcp
+    backendRefs:
+    - group: ""
+      kind: Service
+      name: api-githubcopilot-com
+      namespace: mcp-test
+      port: 443
+EOF
+```
+
+## Step 6: Create the Secret for Authentication
+
+Create a secret containing your GitHub PAT token with the Bearer prefix.
+
+```bash
+# Create the secret with Bearer prefix
+kubectl create secret generic github-token \
+  --namespace=mcp-test \
+  --from-literal=token="Bearer $GITHUB_PAT"
+```
+
+## Step 7: Create the MCPServer Resource
+
+Finally, create the `MCPServer` resource that registers the GitHub MCP server with the gateway:
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: mcp.kagenti.com/v1alpha1
+kind: MCPServer
+metadata:
+  name: github
+  namespace: mcp-test
+spec:
+  toolPrefix: github_
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: github-mcp-external
+  credentialRef:
+    name: github-token
+    key: token
+EOF
+```
+
+## Step 8: Verify
+
+Check that the `MCPServer` is registered:
+
+```bash
+kubectl get mcpservers -n mcp-test
+```
+
+Check the controller logs to see if it discovered the external server:
+
+```bash
+kubectl logs -n mcp-system deployment/mcp-controller | grep github
+```
+
+Verify the broker discovered the tools:
+
+```bash
+kubectl logs -n mcp-system deployment/mcp-broker-router | grep "Discovered.*tools.*github"
+# should show: "Discovered tools mcpURL=https://api.githubcopilot.com:443/mcp #tools=94"
+```
+
+Test the integration through the gateway:
+
+```bash
+# Open MCP Inspector
+make inspect-gateway
+```
+
+This will open the inspector at `http://mcp.127-0-0-1.sslip.io:8888/mcp`
+
+**Verify:**
+
+1. Navigate to **Tools** → **List Tools**
+   - You should see both test server tools and ~90 GitHub tools prefixed with `github_`
+
+2. Test a GitHub tool:
+   - Find and select `github_get_me`
+   - Click **Run Tool**
+   - Should return your GitHub user information
+
+## Cleanup
+
+When done, cleanup things:
+
+```bash
+# Delete the MCPServer and Istio resources
+kubectl delete mcpserver github -n mcp-test
+kubectl delete httproute github-mcp-external -n mcp-test
+kubectl delete service api-githubcopilot-com -n mcp-test
+kubectl delete serviceentry github-mcp-external -n mcp-test
+kubectl delete destinationrule github-mcp-external -n mcp-test
+kubectl delete secret github-token -n mcp-test
+
+# Remove the listener from the Gateway
+kubectl get gateway mcp-gateway -n gateway-system -o json | \
```

Comment:
I wonder if we could make the gateway more permissive by default. It looks like for every external host we support, we would need to add new listeners to the gateway, and when we're done, we remove it. This is more error prone than adding/removing other types of objects like Httproutes.